src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -486,31 +523,18 @@ public inherited sharing class RD2_OpportunityEvaluationService {
      * @return List<RecurringDonation> Recurring Donations with related installment Opportunities and active Schedules
      */
     public List<npe03__Recurring_Donation__c> getRecurringDonations(Set<Id> rdIds) {
-        final String parentAccountField = 'npe03__Contact__r.AccountId';
-
-        Map<String, Schema.DescribeFieldResult> allRdFields = UTIL_Describe.getAllFieldsDescribe(
-            npe03__Recurring_Donation__c.SObjectType.getDescribe().name
-        );
-
-        //Query all fields on the Recurring Donation SObject except CLOB fields.
-        //Thus, there is no need to include fields from the custom field mapping since
-        //they should be already included.
         Set<String> queryFields = new Set<String>();
-        for (String fld : allRdFields.keySet()) {
-            DescribeFieldResult fldResult = allRdFields.get(fld);
-            if (fldResult.getType() == DisplayType.BASE64 ||
-                fldResult.getType() == DisplayType.ENCRYPTEDSTRING ||
-                (fldResult.getSoapType() == SoapType.STRING && fldResult.getLength() > 4096)
-            ) {
-                continue;
-            }
-            queryFields.add(fld);
-        }
 
-        queryFields.add(parentAccountField);
+        queryFields.addAll(rdFields);
+        queryFields.addAll((new RD2_CustomFieldMapper()).getRdFieldNames());
+        queryFields.addAll((new RD2_NamingService()).nameFormatter.fields);
```

Comment:
Quick questions:
* does this still work as expected if automatic naming is not enabled?
* Are there any potential issues with duplicate field names if the fields in the naming format are in a different case?

Reply:
- Yes, if auto naming disable, nameFormatter.fields will return the default value, which is already included in the rdFields. 
- queryFields is a Set<String>, so If there is any duplicate field, the Set will take care of it by default.